### DISCUSSION TITLE
"Infinite Loop" Using Docker in macOS

### DISCUSSION THREAD

**Discussion by peterstory** [mitmproxy/mitmproxy#5794] [General]
I'm hoping someone can explain strange behavior I'm seeing using mitmproxy. I'm running macOS 13.0.1 (Ventura), and attempting to run mitmproxy in Docker with this command:
```
docker run --rm -it \
-v ~/.mitmproxy:/home/mitmproxy/.mitmproxy \
-p 8080:8080 -p 127.0.0.1:8081:8081 \
mitmproxy/mitmproxy mitmproxy
```

When I test mitmproxy with curl, my request appears in mitmproxy as expected:
```
curl --proxy http://127.0.0.1:8080 http://example.com
```

However, if I enable HTTP(S) proxies in macOS settings, then retest using the same curl command, I get some kind of infinite loop, with thousands of entries in mitmproxy that render it unusable. See screenshots for my macOS settings, and the strange mitmproxy output.

![settings1](https://user-images.githubusercontent.com/1322135/206285895-2f0626c8-689d-4a46-a4a8-a7ed64783d2d.png)
![settings2](https://user-images.githubusercontent.com/1322135/206285901-b6c57607-ede9-48a8-ace7-180422778322.png)
![output](https://user-images.githubusercontent.com/1322135/206285902-b6390ef7-fbe8-4ea3-8016-e74e1020de10.png)

**Comment by peterstory** [comment]
I don't see the infinite loop if I run mitmproxy on a separate device from where the proxied curl is run from. However, I'd like to be able to run everything on one device.

**Comment by gekart** [comment]
There actually seems to be a problem on MacOS and Docker. Verified on M1 and Intel (Ventura and Monterey)  https://stackoverflow.com/questions/74817088/running-mitmproxy-on-a-mac-using-docker

**Comment by tsarenkotxt** [comment]
Do you have any updates on this?
